src/Domain/Creators/TimeDependence/ScalingAndZRotation.hpp

```diff
@@ -0,0 +1,198 @@
+        "Expansion values at initial time."};
+  };
+  /// \brief The velocity of the expansion factors.
+  struct Velocity {
+    using type = std::array<double, 2>;
+    static constexpr Options::String help = {"The rate of expansion."};
+  };
+  /// \brief The acceleration of the expansion factors.
+  struct Acceleration {
+    using type = std::array<double, 2>;
+    static constexpr Options::String help = {"The acceleration of expansion."};
+  };
+  /// \brief Whether to use linear scaling or cubic scaling.
+  struct UseLinearScaling {
+    using type = bool;
+    static constexpr Options::String help = {
+        "Whether or not to turn on cubic scaling."};
+  };
+
+  using MapForComposition = detail::generate_coordinate_map_t<
+      tmpl::list<CubicScaleMap,
+                 tmpl::conditional_t<MeshDim == 2, Rotation,
+                                     domain::CoordinateMaps::TimeDependent::
+                                         ProductOf2Maps<Rotation, Identity>>>>;
+
+  using options =
+      tmpl::list<InitialTime, AngularVelocity, OuterBoundary, UseLinearScaling,
+                 InitialExpansion, Velocity, Acceleration>;
+
+  static constexpr Options::String help = {
+      "A spatial radial scaling followed by a rotation about the z-axis.\n"
+      "The spatial radial scaling is either based on a cubic scaling or a\n"
+      "simple linear scaling. If the two expansion functions of time have\n"
+      "the same name then the scaling is a linear radial scaling.\n"
+      "The spatially uniform rotation about is initialized with a\n"
+      "constant angular velocity."};
+
+  ScalingAndZRotation() = default;
+  ~ScalingAndZRotation() override = default;
+  ScalingAndZRotation(const ScalingAndZRotation&) = delete;
+  ScalingAndZRotation(ScalingAndZRotation&&) = default;
+  ScalingAndZRotation& operator=(const ScalingAndZRotation&) =
+      delete;
+  ScalingAndZRotation& operator=(ScalingAndZRotation&&) = default;
+
+  ScalingAndZRotation(double initial_time, double angular_velocity,
+                      double outer_boundary, bool use_linear_scaling,
+                      const std::array<double, 2>& initial_expansion,
+                      const std::array<double, 2>& velocity,
+                      const std::array<double, 2>& acceleration);
+
+  auto get_clone() const -> std::unique_ptr<TimeDependence<MeshDim>> override;
+
+  auto block_maps(size_t number_of_blocks) const
+      -> std::vector<std::unique_ptr<domain::CoordinateMapBase<
+          Frame::Grid, Frame::Inertial, MeshDim>>> override;
+
+  auto functions_of_time(const std::unordered_map<std::string, double>&
+                             initial_expiration_times = {}) const
+      -> std::unordered_map<
+          std::string,
+          std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>> override;
+
+  /// Returns the map for each block to be used in a composition of
+  /// `TimeDependence`s.
+  MapForComposition map_for_composition() const;
```

Comment:
Again, this is probably ok here because it is a composition, but this function is no longer needed in any of the other `TimeDependence`s. I think it's confusing to keep a function "for composition" when there is no more composition.

Reply:
Renamed to `GridToInertialMap`.  (In my next PR there will also be GridToDistortedMap and DistortedToInertialMap).